src/field_theory/adjoin.lean

```diff
@@ -166,81 +259,150 @@ end adjoin_def
 section adjoin_subalgebra_lattice
 variables {F : Type*} [field F] {E : Type*} [field E] [algebra F E] {α : E} {S : set E}
 
-lemma adjoin_eq_bot (h : S ⊆ (⊥ : subalgebra F E)) : adjoin F S = ⊥ :=
-begin
-  rw eq_bot_iff,
-  intros x,
-  rw [subalgebra.mem_coe, subalgebra.mem_coe, algebra.mem_bot],
-  rw algebra.coe_bot at h,
-  apply adjoin_subset_subfield _ _ set.subset.rfl h,
-end
+lemma adjoin_eq_bot (h : S ⊆ (⊥ : intermediate_field F E)) : adjoin F S = ⊥ :=
+eq_bot_iff.mpr ((adjoin_le_iff F S).mpr h)
 
-lemma adjoin_simple_eq_bot (hα : α ∈ ((⊥ : subalgebra F E) : set E)) : F⟮α⟯ = (⊥ : subalgebra F E) :=
+lemma adjoin_simple_eq_bot (hα : α ∈ (⊥ : intermediate_field F E)) : F⟮α⟯ = ⊥ :=
 adjoin_eq_bot (set.singleton_subset_iff.mpr hα)
 
-lemma adjoin_zero : F⟮0⟯ = (⊥ : subalgebra F E) :=
-adjoin_simple_eq_bot (algebra.mem_bot.mpr (is_add_submonoid.zero_mem))
+lemma adjoin_zero : F⟮(0 : E)⟯ = ⊥ :=
+adjoin_simple_eq_bot (⊥ : intermediate_field F E).zero_mem
 
-lemma adjoin_one : F⟮1⟯ = (⊥ : subalgebra F E) :=
-adjoin_simple_eq_bot (algebra.mem_bot.mpr (is_submonoid.one_mem))
+lemma adjoin_one : F⟮(1 : E)⟯ = ⊥ :=
```

Comment:
Idem. If you want, you can add similar lemmas for `bit0`, `bit1`, and elements that are coerced from `nat` and `int`. It's part of building a good API. All of those would make reasonable simp-lemmas, I think.

Reply:
I did it for nat and int. What are you proposing for bit0 and bit1? If F(a)=bot then F(bit0) = bot and similarly for bit1?